src/Microsoft.Framework.Runtime.Abstractions/DiagnosticResult.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Framework.Runtime
+{
+    public class DiagnosticResult
+    {
+        public static readonly DiagnosticResult Successful = new DiagnosticResult(success: true, diagnostics: Enumerable.Empty<DiagnosticMessage>());
+
+        public bool Success { get; }
+
+        public IEnumerable<DiagnosticMessage> Diagnostics { get; }
+
+        public DiagnosticResult(bool success, IEnumerable<DiagnosticMessage> diagnostics)
```

Comment:
I usually follow StyleCop's suggestion. In terms of the order of the elements in a class: http://www.stylecop.com/docs/SA1201.html